src/settings/ReadingRoomAccess/ReadingRoomAccess.js

```diff
@@ -74,49 +24,126 @@ const translations = {
 const ReadingRoomAccess = (props) => {
   const intl = useIntl();
   const stripes = useStripes();
+  const { resources } = props;
+
+  // service points defined in the tenant
+  const servicePoints = _.get(resources, ['RRAServicePoints', 'records', 0, 'servicepoints'], []);
+  /**
+   * A reading room can have more than one service points assigned to it.
+   * but a servicepoint cannot be mapped to more than one reading room
+  */
+  const sps = [];
+  const rrs = _.get(resources, ['values', 'records']);
+  rrs.forEach(rr => {
+    const asp = rr.servicePoints || [];
+    asp.forEach(s => {
+      if (!sps.includes(s.value)) {
+        sps.push(s.value);
+      }
+    });
+  });
+
+  const options = [];
+  servicePoints.forEach(s => {
+    if (!sps.includes(s.id) || s.name === 'None') {
+      options.push({ value: s.id, label: s.name });
+    }
+  });
+
+  const fieldLabels = useMemo(() => ({
+    [readingRoomAccessColumns.NAME]: intl.formatMessage({ id: 'ui-tenant-settings.settings.reading-room-access.name' }),
+    [readingRoomAccessColumns.ISPUBLIC]: intl.formatMessage({ id: 'ui-tenant-settings.settings.reading-room-access.public' }),
+    [readingRoomAccessColumns.SERVICEPOINTS]: intl.formatMessage({ id: 'ui-tenant-settings.settings.reading-room-access.asp' }),
+  }), [intl]);
+
+  const visibleFields = useMemo(() => ([
+    readingRoomAccessColumns.NAME,
+    readingRoomAccessColumns.ISPUBLIC,
+    readingRoomAccessColumns.SERVICEPOINTS,
+  ]), []);
+
+  const getRequiredLabel = useCallback(columnLabel => (
+    <Label required>{columnLabel}</Label>
+  ), []);
+
+  const columnMapping = useMemo(() => ({
+    [readingRoomAccessColumns.NAME]: getRequiredLabel(fieldLabels[readingRoomAccessColumns.NAME]),
+    [readingRoomAccessColumns.ISPUBLIC]: fieldLabels[readingRoomAccessColumns.ISPUBLIC],
+    [readingRoomAccessColumns.SERVICEPOINTS]: getRequiredLabel(fieldLabels[readingRoomAccessColumns.SERVICEPOINTS])
+  }), [fieldLabels, getRequiredLabel]);
 
-  const columnMapping = {
-    name: (
-      <Label
-        tagName="span"
-        required
-      >
-        {
-        intl.formatMessage({ id:'ui-tenant-settings.settings.reading-room-access.name' })
+  const formatter = useMemo(() => getFormatter({ fieldLabels }), [fieldLabels]);
+
+  const validateItem = useCallback((item, items) => {
+    const errors = Object.values(readingRoomAccessColumns).reduce((acc, field) => {
+      const error = getValidators(field)?.(item, items);
+
+      if (error) {
+        acc[field] = error;
       }
-      </Label>),
-    public: intl.formatMessage({ id:'ui-tenant-settings.settings.reading-room-access.public' }),
-    servicePoint: intl.formatMessage({ id:'ui-tenant-settings.settings.reading-room-access.asp' }),
+
+      return acc;
+    }, {});
+
+    return errors;
+  }, []);
+
+  const validate = (item, index, items) => {
+    const itemErrors = validateItem(item, items) || {};
```

Comment:
Items errors looks like redundant, could you please just change to
`return validateItem(item, items) || {}`